flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -43,6 +46,12 @@
  * <p>See {@link TaskExecutorResourceSpec} for details about memory components of TaskExecutor and their relationships.
  */
 public class TaskExecutorResourceUtils {
+	private static final Logger LOG = LoggerFactory.getLogger(TaskExecutorResourceUtils.class);
+
+	private static final String DERIVED_MEMORY_INCONSISTENCE_ERROR_MESSAGE =
+		"The derived shuffle memory size (what is left after subtracting other configured types of memory from " +
+			"the total Flink memory) is inconsistent with its configuration and the configured " +
+			"total flink memory, task heap size and (or) managed memory. ";
```

Comment:
I would not add this message, or at least the part "what is left after subtracting other configured types of memory from the total Flink memory".
I think the sanity check method should not assume how the memory size is derived. It should focus on checking whether the derived value respects all the constraints. In the current way, if later we change the logic how shuffle memory size is derived, this message could be easily overlooked thus providing incorrect information to the user.

Reply:
True, sanity check can be used in other contexts as well. The reason I decided to make it so verbose as there are quite some implicit things going on in background which can be hard to understand. I think this message would improve user experience. I factored out this message to a separate method which wraps the original exception with it. This method is used only in context of configured total Flink and task heap memory:
`sanityCheckShuffleMemoryWithExplicitlySetTotalFlinkAndHeapMemory`.
I believe this message gives more info to the user about the failure context.